ts/output/common/Wrappers/scriptbase.ts

```diff
@@ -215,77 +264,119 @@ export function CommonScriptbaseMixin<
       //
       //  Find the base core
       //
-      let core = this.baseCore = this.childNodes[0];
+      const core = this.baseCore = this.getBaseCore();
       if (!core) return;
-      while (core.childNodes.length === 1 &&
-             (core.node.isKind('mrow') || core.node.isKind('TeXAtom') ||
-              core.node.isKind('mstyle') || core.node.isKind('mpadded') ||
-              core.node.isKind('mphantom') || core.node.isKind('semantics'))) {
-        core = core.childNodes[0];
-        if (!core) return;
-      }
-      if (!('noIC' in core)) return;
-      this.baseCore = core;
       //
       //  Check if the base is a mi or mo that needs italic correction removed
       //
-      if (!(this.constructor as CommonScriptbaseClass).useIC) {
-        (core as CommonMo).noIC = true;
+      if (('noIC' in core) && !(this.constructor as CommonScriptbaseClass).useIC) {
+        (core as unknown as CommonMo).noIC = true;
+      }
+      //
+      // Get information about the base element
+      //
+      this.baseScale = this.getBaseScale();
+      this.baseIc = this.getBaseIc();
+      this.baseIsChar = this.isCharBase();
+    }
+
+    /***************************************************************************/
+    /*
+     *  Methods for information about the core element for the base
+     */
+
+    /**
+     * @return {W}    The wrapper for the base core mi or mo (or whatever)
+     */
+    public getBaseCore(): W {
+      let core = this.getSemanticBase() || this.childNodes[0];
```

Comment:
This confuses me: If the semantic base exists, would that not be a wrapper and therefore I would have take the first child?

Reply:
The wrapper is the output-jax-specific shell around the actual (internal) MathML node, node an enclosing MathML node.  If the fence is an `mo` (an `MmlMo` node), for example, then `getSemanticBase()` will return the `CHTMLmo` object that is the wrapper for that `MmlMo` node.  You don't need to take its children, because it is already the wrapper that you want.